.eslintrc.js

```diff
@@ -379,7 +379,6 @@ module.exports = {
     // function placeholder params are always ignored, and any other unused
     // locals must be justified with a disable comment.
     '@typescript-eslint/no-unused-vars': [ERROR, {ignoreRestSiblings: true}],
-    '@typescript-eslint/prefer-optional-chain': ERROR,
```

Comment:
As much as I love this rule, it needs [linting with type information](https://typescript-eslint.io/getting-started/typed-linting) to be bug-free. I tried adding typed linting but:

* `parserOptions.project` resulted in out-of-memory exceptions
* `EXPERIMENTAL_useProjectService` resulted in 855 files complaining about not being included in the default project (i.e. https://github.com/typescript-eslint/typescript-eslint/pull/9097)

...so my suggestion would be to wait until v8 to use the stabilized [`parserOptions.projectService`](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta#project-service).

Reply:
IIRC, we had a weird type checking setup (to make tests type checked), and I don't remember a good way to deal with it.